Gordon360/Services/RecIM/TeamService.cs

```diff
@@ -0,0 +1,176 @@
+                                .FirstOrDefault(opmt => opmt.MatchID == mt.MatchID
+                                && opmt.TeamID != teamID)
+                                .Score)
+                                .Count(),
+                            Tie = t.MatchTeam
+                                .Where(mt => mt.Match.StatusID == 6
+                                && mt.Score == _context.MatchTeam
+                                .FirstOrDefault(opmt => opmt.MatchID == mt.MatchID
+                                && opmt.TeamID != teamID)
+                                .Score)
+                                .Count(),
+                        }),
+                }).FirstOrDefault();
+            return team;
+        }
+
+        public async Task<int> PostTeamAsync(TeamUploadViewModel t, string username)
+        {
+            var participantID = _context.ACCOUNT
+                                .FirstOrDefault(a => a.AD_Username == username)
+                                .account_id;
+
+            var team = new Team
+            {
+                Name = t.Name,
+                StatusID = t.StatusID,
+                ActivityID = t.ActivityID,
+                Private = t.Private,
+                Logo = t.Logo,
+            };
+            await _context.Team.AddAsync(team);
+            await _context.SaveChangesAsync();
+
+            await AddUserToTeamAsync(team.ID, username);
+            await UpdateParticipantRoleAsync(team.ID, participantID, 5);
+
+            return team.ID;
+        }
+
+        public async Task UpdateParticipantRoleAsync(int teamID, int participantID, int participantRoleID)
+        {
+            var participantTeam = _context.ParticipantTeam.FirstOrDefault(pt => pt.ParticipantID == participantID && pt.TeamID == teamID);
+            participantTeam.RoleType = participantRoleID;
+
+            await _context.SaveChangesAsync();
+        }
+
+        public async Task<int> UpdateTeamAsync(int teamID, TeamPatchViewModel update)
+        {
+            var t = await _context.Team.FindAsync(teamID);
+            t.Name = update.Name ?? t.Name;
+            t.StatusID = update.StatusID ?? t.StatusID;
+            t.Private = (bool)update.Private;
+            t.Logo = update.Logo ?? t.Logo;
+
+            await _context.SaveChangesAsync();
+
+            return teamID;
+        }
+        public async Task AddUserToTeamAsync(int teamID, string username)
+        {
+            var participantID = _context.ACCOUNT
+                                .FirstOrDefault(a => a.AD_Username == username)
+                                .account_id;
+
+            var participantTeam = new ParticipantTeam
+            {
+                TeamID = teamID,
+                ParticipantID = participantID,
+                SignDate = DateTime.Now,
+                RoleType = 3, //default: 3 -> member
+            };
+            await _context.ParticipantTeam.AddAsync(participantTeam);
```

Comment:
Preferably return this value at the end
```suggestion
            var inserted = await _context.ParticipantTeam.AddAsync(participantTeam);
```
... later ...
```
return inserted;
```

Reply:
I just tested and turned out `int.Parse(...)` does not work. So instead I used `toString()` to convert `pt.ParticipantID` to match the string type rather than converting string to int.

Reply:
This is the wrong thread